pkgs/servers/sql/materialize/default.nix

```diff
@@ -0,0 +1,90 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchzip
+, rustPlatform
+, bootstrap_cmds
+, DiskArbitration
+, Foundation
+, cmake
+, libiconv
+, openssl
+, perl
+, pkg-config}:
+
+let
+  fetchNpmPackage = {name, version, sha256, js_prod_file, js_dev_file, ...} @ args:
+  let
+    package = fetchzip {
+      url = "https://registry.npmjs.org/${name}/-/${baseNameOf name}-${version}.tgz";
+      inherit sha256;
+    };
+
+    static = "./src/materialized/src/http/static";
+    cssVendor = "./src/materialized/src/http/static/css/vendor";
+    jsProdVendor = "./src/materialized/src/http/static/js/vendor";
+    jsDevVendor = "./src/materialized/src/http/static-dev/js/vendor";
+
+    files = with args; [
+      { src = js_prod_file; dst = "${jsProdVendor}/${name}.js"; }
+      { src = js_dev_file;  dst = "${jsDevVendor}/${name}.js"; }
+    ] ++ lib.optional (args ? css_file) { src = css_file; dst = "${cssVendor}/${name}.css"; }
+      ++ lib.optional (args ? extra_file) { src = extra_file.src; dst = "${static}/${extra_file.dst}"; };
+  in
+    lib.concatStringsSep "\n" (lib.forEach files ({src, dst}: ''
+      mkdir -p "${dirOf dst}"
+      cp "${package}/${src}" "${dst}"
+    ''));
+
+  npmPackages = import ./npm_deps.nix;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "materialize";
+  version = "0.7.1";
+  rev = "f4bd159fa73d37d44f8ed3f1db13c0c2ff85566f";
+
+  src = fetchFromGitHub {
+    owner = "MaterializeInc";
+    repo = pname;
+    inherit rev;
+    sha256 = "01fhqjzzz9p0gpdbvkfz0p6422akyx5qfqgj3a78mgwiy83jfypj";
+  };
+
+  cargoSha256 = "0gfpv3rqmr18lbja1hl1iipn20fnnxslcwr3l2yb2k06sp19pcz9";
```

Comment:
```suggestion
  cargoSha256 = "sha256-5slgICqLZFqxPymgHvq98BtcD70hfJMr36pvAoQKEJ4=";
```

Reply:
that suggestion doesn't look quite right:

```bash
$ nix to-sri --type sha256 0gfpv3rqmr18lbja1hl1iipn20fnnxslcwr3l2yb2k06sp19pcz9
sha256-6bObwtUGTLG8oCNzRnW31gFhb4yBwqDkoijkivPY1z0=
```

But I went ahead and replaced all the hashes in the derivation with sri ones

Reply:
I didn't mean to replace the hashes with sri ones. I got a hash mismatch and maybe suggested the wrong hash to be updated.